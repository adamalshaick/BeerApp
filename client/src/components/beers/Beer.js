import React, { Component } from "react";
import { BeersColumn } from "../common/styles/StyledColumn";
import PropTypes from "prop-types";
import BeerImage from "./BeerImage";

class Beer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showImage: false
    };
  }
  // onClick = e => {
  //   e.preventDefault();
  //   this.setState({ showImage: true });
  // };

  // onClose = e => {
  //   e.preventDefault();
  //   this.setState({ showImage: false });
  // };

  render() {
    const { beer } = this.props;
    return (
      <BeersColumn className="row">
        <div className="col-8">
          <strong>name:</strong> {beer.name}
          <br />
          <strong>type:</strong> {beer.type}
          <br />
          <strong>price:</strong> {beer.price}
        </div>
        <div className="col-4">
          <BeerImage
            showImage={this.state.showImage}
            beer={beer}
            onClick={() => {
              this.setState(prevState => ({
                showImage: !prevState.showImage
              }));
            }}
          />
        </div>
      </BeersColumn>
    );
  }
}

Beer.propTypes = {
  beer: PropTypes.object.isRequired
};

export default Beer;
