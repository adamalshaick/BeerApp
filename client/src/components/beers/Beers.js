import React, { Component } from "react";
import Beer from "./Beer";
import PropTypes from "prop-types";

export default class Beers extends Component {
  shouldComponentUpdate(nextProps) {
    if (
      nextProps.beers !== this.props.beers ||
      nextProps.sortBy !== this.props.sortBy
    ) {
      return true;
    } else return false;
  }
  render() {
    const { beers } = this.props;
    return beers.map(beer => <Beer beer={beer} />);
  }
}

Beers.propTypes = {
  beers: PropTypes.array.isRequired,
  sortBy: PropTypes.string.isRequired
};
