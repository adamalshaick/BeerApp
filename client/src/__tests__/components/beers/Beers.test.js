import React from "react";
import Beers from "../../../components/beers/Beers";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Beer from "../../../components/beers/Beer";

const Props = {
  beers: [
    {
      beer: "beer 1"
    },
    {
      beer: "beer 2"
    }
  ],
  sortBy: "name"
};

const wrapper = shallow(<Beers {...Props} />);
describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("beers feed", () => {
  it("returns beer for each element in beers array", () => {
    expect(wrapper.find(Beer).length).toEqual(2);
  });
});
