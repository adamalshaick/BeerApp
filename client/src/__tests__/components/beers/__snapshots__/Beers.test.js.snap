// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
Array [
  <Beer
    beer={
      Object {
        "beer": "beer 1",
      }
    }
  />,
  <Beer
    beer={
      Object {
        "beer": "beer 2",
      }
    }
  />,
]
`;
